src/image/loading_displaying.js

```diff
@@ -77,12 +77,35 @@ p5.prototype.loadImage = function(path, successCallback, failureCallback) {
     pImg.drawingContext.drawImage(img, 0, 0);
     pImg.modified = true;
 
-    if (typeof successCallback === 'function') {
-      successCallback(pImg);
+    //GIF
+    if (path.match(/\.[gif]+$/i)) {
+      var req = new XMLHttpRequest();
```

Comment:
`fetch` has support for [`arrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/API/Body/arrayBuffer) as well if you want a syntax more consistent with the loading functions.

Reply:
Cool thanks! I will definitely change that.